src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -911,4 +1158,192 @@ public sealed override int GetHashCode()
+        public IEnumerable<Cci.ICustomAttribute> GetAttributes(EmitContext context) => [];
+        public Cci.ITypeReference GetContainingType(EmitContext context) => ContainingTypeDefinition;
+        public MethodImplAttributes GetImplementationAttributes(EmitContext context) => default;
+        public ISymbolInternal? GetInternalSymbol() => null;
+        public ImmutableArray<Cci.IParameterTypeInformation> GetParameters(EmitContext context)
+            => Parameters.CastArray<Cci.IParameterTypeInformation>();
+        public Cci.IMethodDefinition? GetResolvedMethod(EmitContext context) => this;
+        public IEnumerable<Cci.ICustomAttribute> GetReturnValueAttributes(EmitContext context) => [];
+        public virtual Cci.ITypeReference GetType(EmitContext context) => context.Module.GetPlatformType(Cci.PlatformType.SystemVoid, context);
+        #endregion
+
+        #region IMethodBody
+        public ImmutableArray<Cci.ExceptionHandlerRegion> ExceptionRegions => [];
+        public bool AreLocalsZeroed => false;
+        public bool HasStackalloc => false;
+        public ImmutableArray<Cci.ILocalDefinition> LocalVariables => [];
+        public Cci.IMethodDefinition MethodDefinition => this;
+        public StateMachineMoveNextBodyDebugInfo? MoveNextBodyInfo => null;
+        public ushort MaxStack => _maxStack;
+        public ImmutableArray<byte> IL => _il;
+        public ImmutableArray<Cci.SequencePoint> SequencePoints => [];
+        public bool HasDynamicLocalVariables => false;
+        public ImmutableArray<Cci.LocalScope> LocalScopes => [];
+        public Cci.IImportScope? ImportScope => null;
+        public DebugId MethodId => default;
+        public ImmutableArray<StateMachineHoistedLocalScope> StateMachineHoistedLocalScopes => [];
+        public string? StateMachineTypeName => null;
+        public ImmutableArray<EncHoistedLocalInfo> StateMachineHoistedLocalSlots => [];
+        public ImmutableArray<Cci.ITypeReference?> StateMachineAwaiterSlots => [];
+        public ImmutableArray<EncClosureInfo> ClosureDebugInfo => [];
+        public ImmutableArray<EncLambdaInfo> LambdaDebugInfo => [];
+        public ImmutableArray<LambdaRuntimeRudeEditInfo> OrderedLambdaRuntimeRudeEdits => [];
+        public StateMachineStatesDebugInfo StateMachineStatesDebugInfo => default;
+        public ImmutableArray<SourceSpan> CodeCoverageSpans => [];
+        public bool IsPrimaryConstructor => false;
+        #endregion
+    }
+
+    file sealed class ParameterDefinition(
+        ushort index,
+        string name,
+        Cci.PlatformType type)
+        : ParameterDefinitionBase(index, name)
+    {
+        private readonly Cci.PlatformType _type = type;
+
+        public override Cci.ITypeReference GetType(EmitContext context) => context.Module.GetPlatformType(_type, context);
+    }
+
+    file abstract class ParameterDefinitionBase(
```

Comment:
>ParameterDefinitionBase [](http://example.com/codeflow?start=24&length=23)

Consider whether there is a benefit in extracting and sharing a common base with ```ReturnValueParameter```.